libs/community/langchain_community/retrievers/azure_ai_search.py

```diff
@@ -89,13 +92,21 @@ def format_docs(docs):
     """Name of Azure AI Search service"""
     index_name: str = ""
     """Name of Index inside Azure AI Search service"""
-    api_key: str = ""
+    api_key: Optional[str] = ""
```

Comment:
Optional != NotRequired

Changing to an Optional has no effect since we can base the checks on falsiness

```suggestion
    api_key: Optional[str] = ""
```